.travis.yml

```diff
@@ -1,20 +1,106 @@
+sudo: false
+
 language: rust
-rust:
-- stable
-- beta
-- nightly
-matrix:
-  allow_failure:
-  - rust: nightly
+
 env:
   global:
-    secure: l3/qEC4krRerllLQzni8j5AjngFi6pluWvBWj//1mJLoIEYwxlQ9mYxEdd9BqccWWFn3K0bVYCVC/64+tP6sRfLkZCe2gPUtwe7ITwCDbapUxmkiRObVJCs5yMQZt6idyhHUDKAXKgNCrusfI2BM3tKGBfRK7Cnn/R/7p/U9+q7D1sgJtUKp6ypVzK6A3jLNp3dFLFI19a5KmbZMVsaa7tOhtdDJjjr7ebsc9z7HMW5/OItiWU3FSauVQQlUMaCiEgFuIG7H7OnBAYWB/gNEtLuwfLqU9UjtWk/njNNRnmJ7m3y5HbQhv5H5F5mJUOq9XFlPLwPwyTeVztSGdQm6k8Pp2pgKBUjY27afBl9BWU+msmN6k0oXfhvIebiBPe/x2udiKeFik1xqOOEU1q9dF0sZiuPxCSM1n7tgWklJ8epgaRQaMPPQw9pO/2H5/ynHCJqBlw6WcdiqWtwAyyr/GEx62u/cg5IVkqb7KLmYsWzjS8wYG4CYs1eIxCw2xPZxP0FGuUXvxTBUPipFze6Z7FqxVauXtVe2D7c1P4738HZP660rmR0GYtHtKLny1QxCCK9sxd9JmcezFCSz4YeQ1od9xc0OzGJ2ullKNGizmGfYmgL6X8faNylLIEdaiHAcY16xV3L0g3fXL1Qg360UHQyj7GIv+0nqQnf+H9xRTTU=
-addons:
-  apt:
-    packages:
-    - nodejs
-    - npm
+    - PROJECT_NAME=mdBook
+
+matrix:
+  include:
+    # Stable channel
+    - os: osx
+      env: TARGET=i686-apple-darwin
+      rust: stable
+    - os: linux
+      env: TARGET=i686-unknown-linux-gnu
+      rust: stable
+      addons:
+        apt:
+          packages: &i686_unknown_linux_gnu
+            - gcc-multilib
+    - os: osx
+      env: TARGET=x86_64-apple-darwin
+      rust: stable
+    - os: linux
+      env: TARGET=x86_64-unknown-linux-gnu
+      rust: stable
+      addons:
+        apt:
+          packages:
+            - nodejs
+            - npm
+    - os: linux
+      env: TARGET=x86_64-unknown-linux-musl
+      rust: stable
+    # Beta channel
+    - os: osx
+      env: TARGET=i686-apple-darwin
+      rust: beta
+    - os: linux
+      env: TARGET=i686-unknown-linux-gnu
+      rust: beta
+      addons:
+        apt:
+          packages: *i686_unknown_linux_gnu
+    - os: osx
+      env: TARGET=x86_64-apple-darwin
+      rust: beta
+    - os: linux
+      env: TARGET=x86_64-unknown-linux-gnu
+      rust: beta
+    - os: linux
+      env: TARGET=x86_64-unknown-linux-musl
+      rust: beta
+    # Nightly channel
+    - os: osx
+      env: TARGET=i686-apple-darwin
+      rust: nightly
+    - os: linux
+      env: TARGET=i686-unknown-linux-gnu
+      rust: nightly
+      addons:
+        apt:
+          packages: *i686_unknown_linux_gnu
+    - os: osx
+      env: TARGET=x86_64-apple-darwin
+      rust: nightly
+    - os: linux
+      env: TARGET=x86_64-unknown-linux-gnu
+      rust: nightly
+    - os: linux
+      env: TARGET=x86_64-unknown-linux-musl
+      rust: nightly
+
 install:
-    - npm install stylus nib
+  - sh ci/install.sh
+
+script:
+  - sh ci/script.sh
+
 after_success:
-  - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_BRANCH == "master" && bash deploy.sh
+  - test $TRAVIS_PULL_REQUEST == "false" &&
```

Comment:
I hope this right, this is meant to:

- run on a single build job, and
- the `npm install stylus nib` is only needed for this deploy script

?